src/main/java/com/puppycrawl/tools/checkstyle/BeanPropertyType.java

```diff
@@ -0,0 +1,44 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * The bean property type used in configuration files for clarification of the user friendly type.
+ * For example, some string property may represent a regular expression.
+ *
+ * @noinspection AnnotationClass
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.FIELD)
+public @interface BeanPropertyType {
```

Comment:
Why you think Bean prefix is good in name ?

Reply:
Maybe `ConfigurationPropertyType`?

Reply:
As we do this for XdocsPagesTest
Does it make sense to name it XdocPropertyType?

Reply:
OK, "Xdocs" prefix makes sense.